### PR TITLE
I1114 table of content subpage

### PR DESCRIPTION
**Pull request by DavAug**
See #1114.

A table of contents exists now on every subpage in the dos, as well as a more detailed list in the side bar. Here and there the docstrings were improved by teXing the fomulas.

### CODE REVIEW

File: docs/source/boundaries.rst [hunk 1]
```diff
@@ -9,8 +9,24 @@ Simple boundaries for an optimisation can be created using
...
 implementation of the :class:`Boundaries` interface.
 
+Overview:
+
+- :class:`Boundaries`
+- :class:`LogPDFBoundaries`
+- :class:`RectangularBoundaries`
+
+
+Customisable boundaries
```
Comment (MichaelClerx) [inline docs/source/boundaries.rst:19]: I don't think we should have a sub-header for each class. If the distinction needs to be clearer we could probably modify the CSS instead?

Comment (chonlei) [reply docs/source/boundaries.rst:19]: Agree with Michael for the sub-header for each class -- looks a bit confusing in fact...

Comment (chonlei) [reply docs/source/boundaries.rst:19]: Except if we do this for _every_ autoclass, than we could generate the toctree for all without manually typing in in the summary...?

**Comment by MichaelClerx** [comment]
Maybe this works? https://stackoverflow.com/questions/24129481/how-to-include-a-local-table-of-contents-into-sphinx-doc

**Comment by chonlei** [comment]
> Maybe this works? https://stackoverflow.com/questions/24129481/how-to-include-a-local-table-of-contents-into-sphinx-doc

Unfortunately I think(!) this works only for sub-headers... :-\

**Pull request merged**